ui/apps/platform/src/hooks/useManagedColumns.ts

```diff
@@ -0,0 +1,108 @@
+import * as yup from 'yup';
+import merge from 'lodash/merge';
+
+import useLocalStorage from './useLocalStorage';
+
+// The type of the object stored in local storage
+type TablePreferencesStorage = {
+    columnManagement: Record<string, Record<string, ColumnConfig>>;
+};
+
+// The configuration for a single column in a table
+export type ColumnConfig = {
+    key: string;
+    title: string;
+    isShown: boolean;
+    isShownByDefault: boolean;
+    isUntoggleAble: boolean;
```

Comment:
Tell me more about `isToggleAble` property.
* Has `false` value in tests.
* Not referenced in mutation functions.

Reply:
This is a required property when passing configurations to PatternFly's [ColumnManagementModal](https://www.patternfly.org/extensions/component-groups/column-management-modal). The use case is to display columns in the modal that are in a `disabled` state and cannot have their value changed.

This seems like it would be useful if we want to disallow the toggling of columns based on a runtime value instead of constant like the current implementation is set up. I personally don't think this is something we would ever want to implement, but the PF component requires it, so I added the most minimal implementation. If a use case arises where we would want this, we can adjust the implementation then.